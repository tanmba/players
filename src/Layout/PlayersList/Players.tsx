import React, { useEffect, useState } from "react";
import './index.css';
import {
  homeStyle,
  fieldStyle,
  playerContainerStyle, 
  playerImgStyle, 
  playerInfoStyle,
  title,
  subtitle,
  cardText,
  playerInfoData,
  fontBigCardText
} from "./style";
import Box from "@mui/material/Box";
import TextField from '@mui/material/TextField';
import PlayerDetails from "./PlayerDetails";

/**
 *  Component displaying players card.
 */
const Players = () => {
  const [players, setPlayers] = useState([]);
  const [playerId, setPlayerId] = useState(Number);
  const [errors, setErrors] = useState<any>();
  const [open, setOpen] = useState(false);

  const handleClose = () => setOpen(false);

  const handleOpen = (id: number) => {
    setOpen(true);
    setPlayerId(id)
  };
  
  useEffect(() => {
    const getAPIData = async () => {
      try {
        const apiResponse = await fetch("./data.json");
        const json = await apiResponse.json();
        const data = (json.players);
        setPlayers(data);
      } catch (error) {
        setErrors(error);
        console.error("There was an error!", errors);
      }
    };
    
    getAPIData();
  }, [errors])

  return (
    <Box className="homeBcg" sx={homeStyle}>
      <TextField 
        sx={fieldStyle} 
        label="Rechercher un joueur" 
        variant="outlined" 
      />
      {players
        .sort(function(a: any, b: any){return a.data.rank - b.data.rank})
        .map((player: any, index: number) => (
          <React.Fragment>
            <Box key={index} sx={playerContainerStyle} onClick={() => handleOpen(player.id)}>
              <Box sx={{ width: { xs: "8.5rem", md: "10.5rem" } }}>
                <Box
                  sx={playerImgStyle}
                  component="img" 
                  alt="Player" 
                  src={player.picture} />
              </Box>
              <Box sx={playerInfoStyle}>
                <Box sx={{ ...title, marginBottom: "1.5rem" }}>{player.firstname} {player.lastname}</Box>
                <Box sx={{ display: "flex", justifyContent: "space-between", }}>
                  <Box sx={playerInfoData}>
                    <Box component="span" sx={{ ...subtitle, marginBottom: "0.4rem", }}>Rank</Box>
                    <Box component="span" sx={{ ...cardText, ...fontBigCardText }}>#{player.data.rank}</Box>
                  </Box>
                  <Box sx={playerInfoData}>
                    <Box component="span" sx={{ ...subtitle, marginBottom: "0.4rem", }}>Points</Box>
                    <Box component="span" sx={{ ...cardText, ...fontBigCardText }}>{player.data.points}</Box>
                  </Box>
                  <Box sx={playerInfoData}>
                    <Box component="span" sx={{ ...subtitle, marginBottom: "0.4rem", }}>Country</Box>
                    <Box component="span" sx={{ ...cardText, ...fontBigCardText }}>{player.country.code}</Box>
                  </Box>
                </Box>
              </Box>
            </Box>
            <PlayerDetails id={playerId} players={players} open={open} handleClose={handleClose} />
          </React.Fragment>
        ))
      }
    </Box>
  )
};

export default Players;