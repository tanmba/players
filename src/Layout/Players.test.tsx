import React from 'react';
import { render } from '@testing-library/react';
import Players from "./PlayersList/Players";

describe("Players component", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Players />);

    expect(asFragment()).toMatchSnapshot();
  });
});
