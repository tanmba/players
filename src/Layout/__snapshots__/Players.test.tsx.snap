// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom/Card component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="homeBcg MuiBox-root css-1k6ezy6"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-3k4p6m-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":r0:"
        id=":r0:-label"
      >
        Rechercher un joueur
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1ftyaf0"
          >
            <span>
              Rechercher un joueur
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Players component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="homeBcg MuiBox-root css-1k6ezy6"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-3k4p6m-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":r0:"
        id=":r0:-label"
      >
        Rechercher un joueur
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1ftyaf0"
          >
            <span>
              Rechercher un joueur
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
